Src/Base/AMReX.cpp

```diff
@@ -457,6 +458,23 @@ amrex::Initialize (int& argc, char**& argv, bool build_parm_parse,
     }
 #endif
 
+#if defined(AMREX_USE_MPI) && defined(AMREX_USE_OMP)
+    if (system::verbose > 0) {
+        auto ncores = int(std::thread::hardware_concurrency());
+        if (ncores != 0 && // It might be zero according to the C++ standard.
+            ncores < omp_get_max_threads() * ParallelDescriptor::NProcsPerNode())
+        {
+            amrex::Print(amrex::ErrorStream())
```

Comment:
This is awesome!

The one nitpick that could happen is: the user manages to set `OMP_NUM_THREADS` on the originating batch node only, but forgets to forward the env variable. That way, the warning would not be raised but all but the first node would still have the issue.

`amrex::AllPrint` would check everywhere.

Admittedly, potentially academic and not worth the noise an `AllPrint` would cause.